scanner/debian.go

```diff
@@ -412,25 +414,76 @@ func (o *debian) parseInstalledPackages(stdout string) (models.Packages, models.
 				o.log.Debugf("%s package status is '%c', ignoring", name, packageStatus)
 				continue
 			}
+
 			installed[name] = models.Package{
 				Name:    name,
 				Version: version,
 			}
 
-			if pack, ok := srcPacks[srcName]; ok {
-				pack.AddBinaryName(name)
-				srcPacks[srcName] = pack
-			} else {
-				srcPacks[srcName] = models.SrcPackage{
-					Name:        srcName,
-					Version:     srcVersion,
-					BinaryNames: []string{name},
+			srcPacks = append(srcPacks, models.SrcPackage{
+				Name:        srcName,
+				Version:     srcVersion,
+				BinaryNames: []string{name},
+			})
+
+			if models.IsKernelSourcePackage(o.getDistro().Family, srcName) {
+				switch o.getDistro().Family {
+				case constant.Debian, constant.Raspbian:
+					switch name {
+					case fmt.Sprintf("linux-image-%s", o.Kernel.Release), fmt.Sprintf("linux-headers-%s", o.Kernel.Release):
+						runningKernelSrcPacks = append(runningKernelSrcPacks, models.SrcPackage{
+							Name:    srcName,
+							Version: srcVersion,
+						})
+					default:
```

Comment:
Maybe not needed?

Reply:
It's not strictly necessary, but I wanted to make it clear that it doesn't do anything outside of the case.

Reply:
OK, agree!